File: src/main/resources/webapp/sadues/src/app/service/auth.service.ts
import {Inject, Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {UserLogin, UserRegister} from "../model/user.model";
import {Observable} from "rxjs";
import {SaResponse} from "../model/response/sa-response";
import {AUTH_API_URL} from "../app.constants";

@Injectable({
  providedIn: 'root'
})
export class AuthService {

  constructor(
    @Inject(AUTH_API_URL) private apiUrl: string,
    private http: HttpClient
  ) { }

  register = (userRegister: UserRegister): Observable<SaResponse> => this.http.post<SaResponse>(`${this.apiUrl}/register`, userRegister);
  login = (userLogin: UserLogin): Observable<SaResponse> => this.http.post<SaResponse>(`${this.apiUrl}/login`, userLogin);

   */
}

File: sadues-server/src/main/java/com/elroykanye/sadues/business/service/i/AuthService.java
package com.elroykanye.sadues.business.service.i;

import com.elroykanye.sadues.api.dto.UserDto;
import com.elroykanye.sadues.api.dto.response.SaResponse;

import javax.servlet.http.HttpSession;

public interface AuthService {

    SaResponse register(UserDto.UserRegister userRegister);

    SaResponse login(UserDto.UserLogin userLogin, HttpSession httpSession);
}
